draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,504 +1,528 @@
+   |      | -- { coded packet  ----|      |
+   | FEC  |                        | FEC  |
+   |      | -- repaired packet } --|      |
+   +------+                        +------+
+
+   Inside a host, the CC and FEC entities can be regarded as
+   conceptually separate:
+
+
+
+Kuhn, Lochin, Michel & WeExpires July 11, 2020                  [Page 3]
+
+Internet-Draft           Coding and congestion              January 2020
+
+
+                  |                  --------                  ^
+                  | data             |      | data,            | sending
+                  |                  |      | requirements     | rate (or
+                  v                  |      v                  | window)
+   +-----------------------------+   |    +-------------------------+
+   |             FEC             | <----> |           CC            |
+   +-----------------------------+   |    +-------------------------+
+     ^                   |           |              ^
+     | information       | coded     |              | network
+     | about repaired    | data      |              | measurements
+     | blocks            ------------               |
+
+   As this diagram shows, the inputs to FEC (data to work upon, and
+   signaling from the receiver about losses and/or repaired blocks) are
+   distinct from the inputs to CC. The latter calculates a sending rate
+   or window from network measurements, and it takes the data to send as
+   input, sometimes along with application requirements such as upper/
+   lower rate bounds, periods of quiescence, or a priority.
+
+   It is not clear that the ACK signals feeding into a congestion
+   control algorithm are useful to FEC in their raw form, and vice versa
+   - information about repaired blocks may be quite irrelevant to a CC
+   algorithm.  However, there can be meaningful other interactions
+   (indicated by the horizontal double arrow) between the two entities,
+   usually as a result of their operation rather than by relaying their
+   own raw inputs.  For example, the network measurements carried out by
+   CC can yield a longer-term statistical measure such as a loss ratio
+   which is useful input for a FEC coding scheme.  Similarly, unequal
+   error protection using fountain codes can be used to assign different
+   priorities to blocks of data, and these priorities can be honored by
+   a CC mechanism.
+
+3.  Scope
+
+   This section describes the scope of the document.
+
+3.1.  Type of application
+
+   The document focuses on reliable transmissions.
+
+3.2.  End-to-end
+
+   The document focuses on end-to-end coding, i.e.  cases where coding
+   is added at the server and client end points.  The discussions should
+   then consider fairness with non-coding solutions.
+
+4.  Base solution description
+
+   The solution can be described as follows: [MICHAEL: So far, it hasn't
+   been clear that we're going to describe a "solution".  A solution to

```

Comment:
The idea is to recommend a default behaviour when coding is used at the transport layer (either in the protocol itself like QUIC or as a tunnel e.g. between IP and TCP),.We propose that the client must not hide the loss signal to the sender to avoid the congestion control to have an incomplete or biased information about the network (<-- this is the problem). :-)

Reply:
Well ok, but just saying "don't hide losses for CC" isn't much of a "solution", it's just one recommendation. The issue here is that the document (I think, at least for now) shouldn't be proposing a concrete solution to a problem but instead just discuss the relationship between FEC and CC. So the problem is to call it a "solution".

Reply:
I agree. The scope should be clearer and maybe not focus on providing a solution at the moment. It may end up with recommendations for each case (fec above/in/under cc).